R/bin_utils.R

```diff
@@ -0,0 +1,88 @@
+#' Get Bins of Metadata
+#'
+#' Creates unique age/heights-bins for each record to speed up joining
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset that is the same as that in `derive_params_growth_age()`/`derive_params_growth_height()`
+#'
+#' @param param Character vector
+#'
+#'   Accepted values: "AGE"/"HEIGHT_LENGTH"
+#'
+#' @family internal
+#' @keywords internal
+#' @return list of breaks and labels for age/height bins
+get_bins <- function(meta_criteria, param) {
+  if(param == "AGE") {
+    unique_vals <- sort(unique(meta_criteria$AGE))
+  }
+  else if(param == "HEIGHT_LENGTH") {
+    unique_vals <- sort(unique(meta_criteria$HEIGHT_LENGTH))
+  }
+  breaks <- c(unique_vals, Inf)
+  labels <- paste0(head(breaks, -1), ",", tail(breaks, -1))
+  return(list(breaks = breaks, labels = labels))
+}
+
+
+#' Find closest bin
+#'
+#' @param param Parameter
+#'
+#'  Vector containing parameter that needs closest bin checking
+#'
+#' @param breaks Breaks created in `get_bins()`
+#'
+#' @param labels Labels created in `get_bins()`
+#'
+#' @return detects appropriate bin
+find_closest_bin <- function(param, breaks, labels) {
```

Comment:
this function will help with the rounding say we have L/M/S values for 25 and 26, there will be two unique "bins" created [25, 26) and [26, 27) for values of 25.4 we want it to map to the [25, 26) bin and for 25.5 we want it to map to [26, 27) bin , this function achieves that, agnostic of bin size, e.g say we have bins [1, 2) [2, 3) [3, 10) [10, 11) and the value 9 should map to 10 since its closer to 10 but is technically in the [3, 10) bin it change the bin accordingly